Framework/LiveData/src/StartLiveData.cpp

```diff
@@ -85,6 +70,38 @@ void StartLiveData::init() {
                   "completes.");
 }
 
+/**
+ * After Instrument property is set copy any properties that the instrument's
+ * listener may have to this algorithm.
+ */
+void StartLiveData::afterPropertySet(const std::string &propName) {
+  if (propName == "Instrument") {
+    // Remove old listener's properties
+    const std::vector<Property *> existingProps = getProperties();
+    for (auto existingProp : existingProps) {
+      if (existingProp->getGroup() == listenerPropertyGroup) {
+        removeProperty(existingProp->name());
+      }
+    }
+
+    // Add new listener's properties
+    auto listener = LiveListenerFactory::Instance().create(
```

Comment:
Might be useful to have a comment about this (following) code relating to the  transfer of ownership. Or to put this in a separate named method.